ftw/simplelayout/contenttypes/browser/galleryblock.py

```diff
@@ -13,8 +13,16 @@ class GalleryBlockView(BaseBlock):
     template = ViewPageTemplateFile('templates/galleryblock.pt')
 
     def get_images(self):
-        return self.context.listFolderContents(
-            contentFilter=dict(portal_type=['Image']))
+        imgBrains = self.context.portal_catalog.searchResults(
```

Comment:
@raphael-s if listfoldercontents cant be sorted and you have to give the catalog a path argument. Because taking all objects and then droping the ones which aren't in your context is a horribly inefficient way of doing it.